policy/modules/services/aptcacher.te

```diff
@@ -79,3 +86,27 @@ miscfiles_read_localization(aptcacher_t)
 
 # For some reasons it's trying to mmap /etc/hosts.deny
 sysnet_map_config(aptcacher_t)
+
+#######################################
+#
+# acngtool local policy
+#
+
+allow acngtool_t self:netlink_route_socket r_netlink_socket_perms;
+allow acngtool_t self:tcp_socket create_stream_socket_perms;
+allow acngtool_t self:unix_stream_socket create_stream_socket_perms;
```

Comment:
its not creating `stream` sockets here. Use `create_socket_perms` instead.
It is using this socket to connect to aptcacher_t (via the `aptcacher_unix_stream_connect()` which is currently missing)

Reply:
Not sure about the difference between `create_stream_socket_perms` and `create_socket_perms` TBH, but OK

Reply:
create_stream_socket_perms is a superset of create_socket_perms. It adds the accept and listen permissions. These permissions are used by processes that accept connections through a socket. In this case acngtool_t does not use the unix stream socket to accept connections, instead it uses the unix stream socket to connect to aptcacher. (so aptcacher uses the accept and listen perms to accept connections from the likes of acngtool, but acngtools just needs a unix_stream_socket to connect to aptcache and thus doesnt not listen on a socket or accepts connections

Reply:
If a process maintains a sock_file then, unless its for use with a datagram socket, it is used to allow entities to connectto it with a unix stream socket, and thus the socket needs to be listened on and connections on the socket need to be accepted.